### PR TITLE
Hardening/1306 is double string2double

### PR DESCRIPTION
**Pull request by kzangeli**
New function for isFloat - right now only used by mongoBackend, but to be integrated also in code for the new JSON format in v2.

### CODE REVIEW

File: src/lib/common/string.cpp [hunk 1]
```diff
@@ -718,3 +718,66 @@ std::string servicePathCheck(const char* servicePath)
...
+*   is returned (according to the sign of the value), and ERANGE is stored in errno.
+*   If the correct value would cause underflow, zero is returned and ERANGE is stored in errno.
+*
+* So, we will need to check errno for ERANGE also.
+*
+* This in turn will make a 'HUGE_VAL double' 12e999999 (is that enough? :-)) be treated as an error.
+* It IS a double, but this function will say it is not, as it actually is not a valid double for
+* this computer as the computer cannot represent it as the C builtin type 'double' ... OK!
+*/
+bool isFloat(char* s, double* dP)
```
Comment (fgalan) [inline src/lib/common/string.cpp]: Thus, this functions sums the functionality of a function to check it a string represent a float and to convert the float, right?

I understand that it doesn't make sense to have two separate functions (or maybe I'm wrong?) but, what about of renaming to `str2double`? The rationale is that, from the two functionalities, I think the most "complete" one is the string parsing. Format checking (as the `isFloat` name suggest) is a "side-effect" of the conversion.

Comment (kzangeli) [inline src/lib/common/string.cpp]: Yes, I removed the other one. No sense in having them both.
And no, I am not very convinced that 'isFloat' is the best possible name. Far from it.
str2double is just so ugly ... but, yes, it describes better what the function actually does.

I was thinking of atod also (atof and atoi already exists bue atod doesn't)

Comment (fgalan) [inline src/lib/common/string.cpp]: str2double is ugly but at the same time describes clearly the functionality :)

Another suggestion: toDouble()

Comment (fgalan) [inline src/lib/common/string.cpp]: ...although `toDouble()` may lead to think that the return value of the function is a double (and it is the bool that means sucess/failure). Forget it :)

File: src/lib/common/string.h [hunk 2]
```diff
@@ -161,16 +161,26 @@ extern void strReplace
...
+
+
+
 /* ****************************************************************************
 *
 * toString -
 */
-template <typename T>
-std::string toString(T t) {
+template <typename T> std::string toString(T t)
```
Comment (fgalan) [inline src/lib/common/string.h]: In which part of the code is this function used?

File: src/lib/common/string.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/common/string.cpp]: > Yes, I removed the other one. No sense in having them both.

You mean in this PR? There is no trace in the diff of that removal... or at least I'm not able to find it.

File: src/lib/common/string.h (hunk 2 above)
Comment (kzangeli) [inline src/lib/common/string.h]: this template is part of the new json render proposal

File: src/lib/common/string.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/common/string.cpp]: Yes, I had two functions initially, but I decided to remove one ...
New name: str2double

File: src/lib/common/string.h (hunk 2 above)
Comment (fgalan) [inline src/lib/common/string.h]: NTC

File: src/lib/common/string.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/common/string.cpp]: Name changed to str2double in e215979
Removing one of the two functions might have been done before I even sent the PR, that could be why you can't see it ...

**Comment by fgalan** [comment]
LGTM

**Pull request merged**